### ISSUE TITLE
Example of training LoLA

### ISSUE THREAD

**Issue by shirayu** [kohya-ss/sd-scripts#85]
Hi, thank you for nice work.

I have been trying to train LoRA, but have not yet succeeded.
To be precise, the loss goes down and `last.safetensors` is generated, but using them hardly changes the image produced [like this](https://github.com/shirayu/example_lora_training/issues/1).

I have tried for days with different parameters, but have not been able to identify the problem.

I uploaded the training data and scripts at https://github.com/shirayu/example_lora_training .
Any advice would be appreciated.

The command for training: https://github.com/shirayu/example_lora_training/blob/cdf08770e41d0cf82ee5c7e20dc1dfaed8ea824b/train_lora.zsh

**Comment by shirayu** [comment]
Training without captions might not be able to work with these images.
I'll try it with captions by WD14Tagger.

**Comment by shirayu** [comment]
Thanks for your advice!

I followed your advice and set the parameters.
By setting parameters that are over-learning, it is now possible to generate slightly altered images.
However, I have tried many combinations of parameters and have yet to get it right.

As noted [here](https://wikiwiki.jp/sd_toshiaki/LoRA), we may need captions edited in a special way to learn characters.
I will investigate further on another day.

**Comment by shirayu** [comment]
Thank you for your kind comments!

I updated to sd-scripts v0.4.0 and tried again with some settings but had no changes.
I also tried using frog images described [the post](https://note.com/kohya_ss/n/nba4eceaa4594) , but trainings were not successful.



I am sorry to trouble you, but could you please provide me with any input/output data and the training command of LoRA training?
I would like to reproduce the training with exactly the same settings.

**Comment by shirayu** [comment]
Now, I'm trying [train_dreambooth_lora.py](https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora.py) in diffusers.
That seems like working well.
I will write more details later.

**Comment by shirayu** [comment]
Using that data and the training commands the loss reached NaN.
There may be a problem with the dependent packages I installed.
I will investigate further.

**Comment by shirayu** [comment]
Got it!!!

I have used xFormers `0.0.15+e163309.d20221231`, and got `NaN`.
But with `0.0.15.dev0+1b1fd8a.d20230125` which is mentioned [here](https://github.com/facebookresearch/xformers/issues/631#issuecomment-1398358330), I finally succeeded in the test!

I will find the most stable xFormers.

Thank you for your support!

**Comment by thr3a** [comment]
@shirayu 

Thanks for sharing xformars problem solving!
Learning with [sample data](https://note.com/kohya_ss/n/nb20c5187e15a) also worked with xformers 0.0.17.dev447.

**Comment by litaotju** [comment]
It's a known issue of xformers library on 30x GPUs, try install the dev version of the xformers.
This works on my 3060.

pip install xformers==0.0.17.dev466
See https://github.com/facebookresearch/xformers/issues/631

**Comment by ambahk** [comment]
In the examples, I see a number mean?
1_girl
5_Wikipe-pan
20_usu frog
1_frog

thanks
